contracts/token/ERC20.cairo

```diff
@@ -98,15 +98,26 @@ func _mint{
         pedersen_ptr : HashBuiltin*,
         range_check_ptr
     }(recipient: felt, amount: Uint256):
-    
+    alloc_locals
+    assert_not_zero(recipient)
+
     let (balance: Uint256) = balances.read(user=recipient)
     # the underscore is for the 1 bit carry
     let (new_balance, _: Uint256) = uint256_add(balance, amount)
     balances.write(recipient, new_balance)
 
-    let (supply: Uint256) = total_supply.read()
+    let (local supply: Uint256) = total_supply.read()
     # the underscore is for the 1 bit carry
-    let (new_supply, _: Uint256) = uint256_add(supply, amount)
+    let (local new_supply, _: Uint256) = uint256_add(supply, amount)
+
+    # reassign syscall_ptr and pedersen_ptr to avoid revocation
+    local syscall_ptr: felt* = syscall_ptr
+    local pedersen_ptr: HashBuiltin* = pedersen_ptr
+
+    # overflow check
+    let (enough_balance) = uint256_lt(supply, new_supply)
+    assert_not_zero(enough_balance)
```

Comment:
```suggestion
    let (is_overflow) = uint256_lt(supply, new_supply)
    assert_not_zero(is_overflow)
```
This has nothing to do with balances. Moreover, I think we can use the above carry to check for overflows (the second return value from `uint256_add` dropped into `_`)

Reply:
Great point! The code looks much cleaner now :)